webapp/widgets/market_dashboard/views/show.ejs

```diff
@@ -0,0 +1,506 @@
+        </div>
+      </div>
+      <!-- end of row -->
+      <div class="row">
+        <div class="col-md-6 col-12 ">
+          <div class="card card-dark mb-2">
+            <div class="card-body d-flex flex-column">
+              <div class="card-title fs-6">Session Income 🤩</div>
+              <canvas
+                id="session-income"
+                class="card code px-2 py-1"
+                height="255"
+              >
+              </canvas>
+            </div>
+          </div>
+        </div>
+
+        <div class="col-md-6 col-12 ">
+          <div class="card card-dark mb-2">
+            <div class="card-body d-flex flex-column">
+              <div class="card-title fs-6">Session Payout 🤔</div>
+              <canvas
+                id="session-payout"
+                class="card code px-2 py-1"
+                height="255"
+              >
+              </canvas>
+            </div>
+          </div>
+        </div>
+      </div>
+      <!-- end of row -->
+    </div>
+
+    <script>
+        const incomeChart = initTimeChart("Income", "session-income", "rgb(46, 204, 113)")
+        const payoutChart = initTimeChart("Payout", "session-payout", "rgb(231, 76, 60)")
+        let { min_transaction, min_session_dataset, rate_txn_icon, rate_text, rate_color } = <%- JSON.stringify(storage) %>;
+        const MIN_TRASACTION = parseInt(min_transaction)
+        const MIN_SESSION_DATASET = parseInt(min_session_dataset)
+        let txns = []
+        let previousWinners = []
+        let previousRichestPlayers = []
+
+        function show({key, data}) {
+          if(!data) return
+          if(key === "STATUS"){
+            const {marketOpen} = data
+            $("#market_open").removeClass()
+            $("#market_open").addClass("animate__animated animate__fadeIn")
+            if(marketOpen){
+              $("#market_open").addClass("status online")
+            } else {
+              $("#market_open").addClass("status offline")
+            }
+          }
+          else if (key === "TRANSACTION"){
+            const {username, timestamp,  amount, gain, rate, txnTime} = data
+            if(txns.length === 0){
+              $("#transaction").empty()
+            }
+
+            if(txns.length >= MIN_TRASACTION){
+              $(`#txn-${txns[0].timestamp}`).removeClass()
+              $(`#txn-${txns[0].timestamp}`).addClass("animate__animated animate__fadeOutUp ")
+              setTimeout(() => {
+                $(`#txn-${txns[0].timestamp}`).remove()
+                txns.shift()
+              },300)
+            }
+            txns.push(data)
+            let message = displayGachaTransaction(username, amount, gain, rate, txnTime)
+            $("#transaction").append(`<li id="txn-${timestamp}" class="animate__animated animate__fadeInUp" >${message}</li>`)
+          } else if (key === "LATEST_WINNERS"){
+            const {winners} = data
+            if(_.isEqual(previousWinners, winners)) return
```

Comment:
I found lodash again at line 317, 334, 434

Reply:
sorry again, i think it use from module forgot about frontend 😢🙏,

Reply:
No worries 😸